### PR TITLE
Upgraded to .Net 9, including Assets, and fixed theming issue

### PR DESCRIPTION
**Pull request by coderdnewbie**
Upgraded to .Net 9, including Assets, and fixed theming issue.

### CODE REVIEW

File: src/DemoFluentUI/Components/DarkModeFluentSwitch.razor [hunk 1]
```diff
@@ -0,0 +1,37 @@
+﻿<FluentDesignTheme @bind-Mode="@Mode"
```
Comment (dvoituron) [inline src/DemoFluentUI/Components/DarkModeFluentSwitch.razor]: I agree with you to upgrade to .NET9 but not to change the code about the Theme Switcher. The goal is to explain like in the demo made by Vincent during the Show, not to have all features. For that, we have the doc website and the doc source code. Keep simple os better 😀

**Review by dvoituron — CHANGES_REQUESTED** [review]

File: src/DemoFluentUI/Components/DarkModeFluentSwitch.razor (hunk 1 above)
Comment (coderdnewbie) [reply src/DemoFluentUI/Components/DarkModeFluentSwitch.razor]: The demo made by Vincent is flawed, if you look at the console log you will see the following error, which may put people off using fluentui - 'Cannot read properties of null (reading 'mode')'. 

If you want reject this pull request, I can make another one with just the .Net 9 upgrade changes, but I would prefer if we had good theming without any errors.

failure

File: src/DemoFluentUI/Components/App.razor [hunk 2]
```diff
@@ -5,13 +5,19 @@
...
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <base href="/" />
-    <link rel="stylesheet" href="app.css" />
-    <link rel="stylesheet" href="DemoFluentUI.styles.css" />
+    <link rel="stylesheet" href="@Assets["app.css"]" />
+    <link rel="stylesheet" href="@Assets["DemoFluentUI.styles.css"]" />
+    <ImportMap />
     <link rel="icon" type="image/x-icon" href="favicon.ico" />
     <HeadOutlet />
+    <script src="_content/Microsoft.FluentUI.AspNetCore.Components/Microsoft.FluentUI.AspNetCore.Components.lib.module.js" type="module" async></script>
```
Comment (vnbaaij) [inline src/DemoFluentUI/Components/App.razor]: This is not needed. The script will be added by the library automaticlly

File: src/DemoFluentUI/Components/DarkModeFluentSwitch.razor (hunk 1 above)
Comment (vnbaaij) [reply src/DemoFluentUI/Components/DarkModeFluentSwitch.razor]: Yes, it is flawed but it is ok for this usecase

**Review by vnbaaij — CHANGES_REQUESTED** [review]

**Comment by coderdnewbie** [comment]
Fine, then please reject this pull request and I will do the .Net9 changes only tomorrow. Thanks.

**Comment by coderdnewbie** [comment]
Upgrade to .Net 9 -> 9a58d3f has only the dotnet 9.0 changes, please look at the files changed.

**Pull request merged**